projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php

```diff
@@ -781,15 +781,22 @@ function wpcom_site_has_global_styles_in_personal_plan( $blog_id = 0 ) {
 		return false;
 	}
 
-	if ( ! function_exists( '\ExPlat\assign_given_user' ) ) {
+	if ( ! function_exists( '\ExPlat\assign_given_user' ) || ! function_exists( '\WPCOM\Experiments\Internal\get_cached_experiment_by_name' ) ) {
+		return false;
+	}
+
+	// If the experiment is not running and the request isn't proxied, we can return early.
+	$exp     = \WPCOM\Experiments\Internal\get_cached_experiment_by_name( 'calypso_plans_global_styles_personal_20240127' );
+	$proxied = function_exists( 'wpcom_is_proxied_request' ) ? wpcom_is_proxied_request() : false;
```

Comment:
We should use A8C_PROXIED_REQUEST constant since the result is already computed on every request. Calling the function directly will re-execute the code every time (going through an array).